mosec/shm.py

```diff
@@ -0,0 +1,67 @@
+import pickle
+from typing import Any, Callable, List
+
+try:
+    from pyarrow import plasma  # type: ignore
+except ImportError:
+    """
+    We don't include pyarrow as our dependency.
+    Users should install it as third party to enable
+    the shared memory IPC feature.
+    """
+
+
+class ShmClient:
+    def __init__(self, shm_path: str = "") -> None:
+        """Initialize a client which may utilize plasma as shared memory.
+
+        Args:
+            shm_path (str, optional): path of the plasma server. Defaults to "".
+        """
+        self.client = plasma.connect(shm_path)
+
+    def put(self, data: Any) -> plasma.ObjectID:
+        """Individual put"""
+        return self.client.put(data)
+
+    def get(self, object_ids: List[plasma.ObjectID]) -> List[Any]:
+        """Batch get"""
+        objects = self.client.get(object_ids)
+        self.client.delete(object_ids)
+        return objects
+
+
+class ShmWrapper:
+    def __init__(
+        self,
+        shm_cli: ShmClient,
+        serializer: Callable = pickle.dumps,
+        deserializer: Callable = pickle.loads,
+    ) -> None:
```

Comment:
Can but what's the benefit? I think `ShmClient` is just the client, not necessarily need to contain the wrapper functionality.

Reply:
They belong to the same module. Why do we need to maintain two classes for the same purpose? I don't think it's necessary to call two one-line functions in another related class function.

Reply:
Not about one line or two line. They serve different purposes. Client is general, and as the name suggests it's just a client, nothing related to any other logic. Warper is specially for wrapping the decode/encode so that it can use the client to talk with shm server. They are of course related but still reasonable to separate. Of course they can also be put together. Not a big deal, and I still prefer to separate because this makes each class's purpose clearer.